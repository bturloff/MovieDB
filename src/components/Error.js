import React from "react";

const Error = ({ title, body }) => {
  return (
    <div className="error card">
      <div>
        <i className="material-icons small flow-text">error</i>
        <h3 className="flow-text">{title}</h3>
      </div>
      <p>{body}</p>
    </div>
  );
};

export default Error;
