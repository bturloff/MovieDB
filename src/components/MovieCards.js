import React from "react";
import LazyLoad from "react-lazyload";
const posterURL = "https://image.tmdb.org/t/p/w200/";

const MovieCards = ({ movies, onClick }) => {
  return (
    <div className="row movie-grid">
      {movies.map(m => (
        <div className="movie-card col s12 m6 l3" key={m.id}>
          {!m.poster_path ? (
            <div className="placeholder card" />
          ) : (
            <LazyLoad height={"100%"} offset={-200}>
              <div className="img-wrapper" onClick={() => onClick(m)}>
                <img
                  className="card-image animated zoomIn card"
                  src={`${posterURL + m.poster_path}`}
                  alt=""
                />
              </div>
            </LazyLoad>
          )}
        </div>
      ))}
    </div>
  );
};

export default MovieCards;
