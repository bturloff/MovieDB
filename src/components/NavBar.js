import React from "react";

const NavBar = () => {
  return (
    <div className={`navbar-fixed navbar`}>
      <nav className="blue-grey darken-4">
        <div className="nav-wrapper container">
          <span className="brand-logo">MovieDB</span>
        </div>
      </nav>
    </div>
  );
};

export default NavBar;
