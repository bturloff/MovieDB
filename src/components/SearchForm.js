import React from "react";

function SearchForm({ onSubmit, onChange, value }) {
  return (
    <form className="searchForm" onSubmit={onSubmit}>
      <div className="input-field">
        <input
          placeholder="i.e. Bourne Identity"
          id="search"
          type="text"
          className="validate"
          value={value}
          onChange={onChange}
        />
        <label htmlFor="search">Search for a movie</label>
        <i className="material-icons search-button" onClick={onSubmit}>
          search
        </i>
      </div>
    </form>
  );
}

export default SearchForm;
