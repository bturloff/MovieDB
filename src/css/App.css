.color1 {
  background-color: #14181e;
}

.App {
  min-height: 100vh;
  /* Permalink - use to edit and share this gradient: https://colorzilla.com/gradient-editor/#263238+0,455a64+79 */
  background: #263238; /* Old browsers */
  background: -moz-linear-gradient(top, #263238 0%, #455a64 79%); /* FF3.6-15 */
  background: -webkit-linear-gradient(
    top,
    #263238 0%,
    #455a64 79%
  ); /* Chrome10-25,Safari5.1-6 */
  background: linear-gradient(
    to bottom,
    #263238 0%,
    #455a64 79%
  ); /* W3C, IE10+, FF16+, Chrome26+, Opera12+, Safari7+ */
  filter: progid:DXImageTransform.Microsoft.gradient( startColorstr='#263238', endColorstr='#455a64',GradientType=0 ); /* IE6-9 */
}

p {
  font-size: 1.4rem;
}

input {
  color: white;
}

.input-field {
  color: #4caf50;
}

.navbar {
  margin-bottom: 50px;
}

.searchForm {
  max-width: 900px;
  margin: auto;
}

.searchForm i {
  cursor: pointer;
}

.searchForm > div {
  display: flex;
  align-items: center;
}

.searchForm > div > i {
  position: absolute;
  right: 0;
}

.movie-card {
  /* border-radius: 8px; */
  /* height: 200px; */
  /* width: 240px; */
  /* overflow: hidden; */
  height: 300px;
  /* background-color: blue; */
  /* width: 25%; */
  margin-bottom: 20px;
  /* margin: 0 20px 20px 20px; */
  /* border: 2px solid red; */
  /* border-radius: 10px; */
}

@media only screen and (max-width: 800px) {
  .card-image {
    object-fit: contain;
  }
}

.card-image {
  width: 100%;
  height: 300px;
  /* object-fit: contain; */
  border-radius: 8px;
  cursor: pointer;
}

.movie-grid {
  /* display: flex;
  flex-wrap: wrap; */
}

.placeholder {
  height: 300px;
  width: 100%;
  background-color: lightgray;
  border-radius: 8px;
}

.main {
  max-width: 900px;
  margin: auto;
}
.main > .row {
  margin-bottom: 0;
}

.modal-image-wrapper {
  width: 800px;
  height: 450px;
}

.mine {
  width: 800px;
  height: 450px;
  /* you need to match the shadow color to your background or image border for the desired effect*/
  /* box-shadow: 0 0 35px 35px white inset; */
  /* box-shadow: 0 0 50px 50px white inset; */
  /* border-radius: 75px; */
  background-position: center;
  background-size: cover;
  background-repeat: no-repeat;
}

.modal-content {
  display: flex;
  width: 100%;
  position: relative;
  bottom: 0;
  left: 0;
  min-height: 360px;
}
.vertical-center {
  margin: 0;
  position: absolute;
  top: 50%;
  left: 50%;
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.modal-content .image-wrapper {
  position: relative;
}

.modal-content img {
  display: block;
  margin: auto;
}

.modal-content div:first-child {
  width: 250px;
}
.modal-content div:nth-child(2) {
  padding: 0 20px;
  width: 550px;
}

.error {
  background-color: lightyellow;
  margin: 60px;
  padding: 60px;
}

.error h3 {
  margin: 20px 0 20px 10px;
}

.error p {
  margin-left: 40px;
}

.error div {
  display: flex;
  align-items: center;
}

.modal-footer {
}
.modal-footer span {
  float: right;
  margin: 0 20px 20px 0;
}
