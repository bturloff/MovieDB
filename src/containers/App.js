import React from "react";
import "../css/App.css";
import "../css/animate.css";
import NavBar from "../components/NavBar";
import SearchForm from "../components/SearchForm";
import { connect } from "react-redux";
import MovieCards from "../components/MovieCards";
import InfiniteScroll from "react-infinite-scroller";

import Error from "../components/Error";
import MovieModal from "../components/MovieModal";
import { fetchMovies, clearMovies } from "../actions/index";

export class App extends React.Component {
  state = {
    movies: [],
    modalIsOpen: false,
    movieSelected: null,
    searchField: ""
  };

  /**
   * Opens Modal when user selects a movie card
   */
  _handleCardClick = movieSelected => {
    this.setState({ modalIsOpen: true, movieSelected });
  };

  /**
   * Close movie modal
   */
  _closeModal = () => {
    this.setState({ modalIsOpen: false });
  };

  /**
   * fetch movies on searchForm submit
   */
  _handleSubmit = e => {
    const { dispatch } = this.props;
    dispatch(clearMovies());
    dispatch(fetchMovies(this.state.searchField));
    e.preventDefault();
  };

  /**
   * Update state as user types in search field
   */
  _handleChange = e => {
    this.setState({ searchField: e.target.value });
  };
  /**
   * Fetch more movies when user scrolls to bottom of screen
   */
  fetchMoreMovies = page => {
    this.props.dispatch(fetchMovies(this.state.searchField, page));
  };

  render() {
    const movies = this.props.movies || [];
    const { movieSelected, modalIsOpen, searchField } = this.state;
    const { error } = this.props;
    return (
      <div className={`App`}>
        <NavBar />
        <MovieModal
          movie={movieSelected}
          isOpen={modalIsOpen}
          closeModal={this._closeModal}
        />
        <SearchForm
          onSubmit={this._handleSubmit}
          onChange={this._handleChange}
          value={searchField}
        />
        <div className="main">
          {error ? (
            <Error title="Something went wrong" body="Please try again later" />
          ) : (
            <InfiniteScroll
              pageStart={0}
              loadMore={this.fetchMoreMovies}
              hasMore={true}
            >
              <MovieCards movies={movies} onClick={this._handleCardClick} />
            </InfiniteScroll>
          )}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    movies: state.movies,
    error: state.fetchError
  };
};

export default connect(mapStateToProps)(App);
