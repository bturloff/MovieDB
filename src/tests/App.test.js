import React from "react";
import ReactDOM from "react-dom";
import { App } from "../containers/App";
import Adapter from "enzyme-adapter-react-16";
import mock from "./mock.json";

import Enzyme, { shallow } from "enzyme";
import MovieCards from "../components/MovieCards";
import SearchForm from "../components/SearchForm";
import NavBar from "../components/NavBar";
import Error from "../components/Error";

Enzyme.configure({ adapter: new Adapter() });

test("App renders without crashing", () => {
  shallow(<App />);
});

/**
 * MovieCards
 * */
test("MovieCards renders without crashing", () => {
  shallow(<MovieCards movies={[]} />);
});
test("MovieCards renders correctly with movies array", () => {
  const onClick = jest.fn();
  const wrapper = shallow(<MovieCards movies={[]} onClick={onClick} />);
  expect(wrapper).toMatchSnapshot();
  wrapper.setProps({ movies: mock.movies });
  expect(wrapper).toMatchSnapshot();
  wrapper
    .find(".img-wrapper")
    .first()
    .simulate("click");
  expect(onClick.mock.calls.length).toBe(1);
});

/**
 * SearchForm
 */
test("Search renders", () => {
  const onSubmit = jest.fn();
  const onChange = jest.fn();
  const value = "hello you";
  const wrapper = shallow(
    <SearchForm onSubmit={onSubmit} onChange={onChange} value={value} />
  );

  /** verify input value */
  const inputWrapper = wrapper.find("input").first();
  expect(inputWrapper.props().value).toEqual("hello you");

  /** verify onChange execution upon updating form */
  inputWrapper.simulate("change");
  expect(onChange).toBeCalled();

  /** verify submit button execution */
  wrapper
    .find(".search-button")
    .first()
    .simulate("click");
  expect(onSubmit).toBeCalled();
});

/**
 * NavBar
 */
test("Navbar should match snapshot", () => {
  const wrapper = shallow(<NavBar />);
  expect(wrapper).toMatchSnapshot();
});

/**
 * Error
 */
test("Error should render correctly", () => {
  const wrapper = shallow(
    <Error title="Important Error" body="come back later" />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.contains("Important Error")).toBeTruthy();
});
