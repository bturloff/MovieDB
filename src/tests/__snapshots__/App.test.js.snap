// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render correctly 1`] = `ShallowWrapper {}`;

exports[`MovieCards renders correctly with movies array 1`] = `ShallowWrapper {}`;

exports[`MovieCards renders correctly with movies array 2`] = `ShallowWrapper {}`;

exports[`Navbar should match snapshot 1`] = `ShallowWrapper {}`;
